automatic/puppet-bolt/tools/chocolateyinstall.ps1

```diff
@@ -1,26 +1,33 @@
-﻿$packageName = 'puppet-bolt'
-$url32       = ''
-$url64       = 'https://downloads.puppet.com/windows/puppet-tools/puppet-bolt-3.5.0-x64.msi'
-$filename32  = ''
-$filename64  = 'puppet-bolt-3.5.0-x64_x64.msi'
-$checksum32  = ''
-$checksum64  = 'FBBF91F0EFEBC61F24D0AE4FF067E6504624406ADE960727862EE74C61B29CE1'
-
+﻿$ErrorActionPreference = 'Stop';
 $toolsDir = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"
+$fileLocation = Join-Path $toolsDir 'puppet-bolt-3.5.0-x64_x64.msi'
+$helpers = Join-Path $toolsDir 'helpers.ps1'
 
-$packageArgs = @{
-  packageName    = $packageName
-  fileType       = 'MSI'
-  silentArgs     = "/qn /norestart"
-  validExitCodes = @(0, 3010, 1641)
-}
+. $helpers
+
+$pp = Get-PackageParameters
 
-if ([string]::IsNullOrEmpty($filename32)) {
-  # If 64bit only, then only use the file parameter
-  $packageArgs['file'] = (Join-Path -Path $toolsDir -ChildPath $filename64)
-} else {
-  $packageArgs['file'] = (Join-Path -Path $toolsDir -ChildPath $filename32)
-  $packageArgs['file64'] = (Join-Path -Path $toolsDir -ChildPath $filename64)
+$packageArgs = @{
+  packageName    = $env:ChocolateyPackageName
+  softwareName   = 'Puppet Bolt*'
+  file           = $fileLocation
+  fileType       = 'msi'
+  silentArgs     = "/qn /norestart /l*v `"$env:TEMP\$env:ChocolateyPackageName.$env:ChocolateyPackageVersion.log`""
+  validExitCodes = @(0, 1641, 3010)
+  checksum       = 'FBBF91F0EFEBC61F24D0AE4FF067E6504624406ADE960727862EE74C61B29CE1'
+  checksumType   = 'sha256'
+  destination    = $toolsDir
 }
 
 Install-ChocolateyInstallPackage @packageArgs
+
+#Put Bolt on the Machine Path, as opposed to User. This is good for C4B customers running Background Mode, as the User PATH is in Agent context, causing issues.
+if ($pp['InstallOnMachinePath']) {
+  Remove-PathVariable -VariableToRemove "C:\Program Files\Puppet Labs\Bolt\bin"

```

Comment:
The user can choose to install to any drive or folder, how do we handle `C:\Program Files\`?

Reply:
💯 Program files is redirectable and should never be hardcoded. And is complicated on 32 v 64 bit program files.  MSI handles this natively.